pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -585,6 +585,19 @@ public static void addColumnMetadataInfo(PropertiesConfiguration properties, Str
         String.valueOf(columnIndexCreationInfo.getTotalNumberOfEntries()));
     properties.setProperty(getKeyFor(column, IS_AUTO_GENERATED),
         String.valueOf(columnIndexCreationInfo.isAutoGenerated()));
+    FieldSpec.MaxLengthExceedStrategy maxLengthExceedStrategy = fieldSpec.getMaxLengthExceedStrategy();
+    if (dataType.equals(DataType.STRING)) {
```

Comment:
We want to set it for all var-length types so that the schema can be properly re-constructed.
For `SCHEMA_MAX_LENGTH_EXCEED_STRATEGY`, we want to add it when it is configured

Reply:
Sounds good to me! Updated accordingly!